src/server/routes/welcome.ts

```diff
@@ -273,6 +288,203 @@ router.get(
 	},
 );
 
+router.get(
+	'/welcome/review',
+	loginMiddlewareOAuth,
+	handleAsyncErrors(async (req: Request, res: ResponseWithRequestState) => {
+		// eslint-disable-next-line functional/no-let
+		let state = res.locals;
+		const sc_gu_u = req.cookies.SC_GU_U;
+
+		try {
+			const consentsData = {
+				consents: await getUserConsentsForPage({
+					pageConsents: CONSENTS_DATA_PAGE,
+					ip: req.ip,
+					sc_gu_u,
+					request_id: res.locals.requestId,
+					accessToken: res.locals.oauthState.accessToken?.toString(),
+				}),
+			};
+
+			state = mergeRequestState(state, {
+				pageData: {
+					...consentsData,
+				},
+			} as RequestState);
+
+			trackMetric('NewAccountReview::Success');
+		} catch (error) {
+			logger.error(`${req.method} ${req.originalUrl}  Error`, error, {
+				request_id: res.locals.requestId,
+			});
+
+			const { message } = error instanceof ApiError ? error : new ApiError();
+
+			state = mergeRequestState(state, {
+				globalMessage: {
+					error: message,
+				},
+			});
+
+			trackMetric('NewAccountReview::Failure');
+		}
+
+		const html = renderer('/welcome/review', {
+			pageTitle: 'Your data',
+			requestState: state,
+		});
+		return res.type('html').send(html);
+	}),
+);
+
+router.get(
+	'/welcome/newsletters',
+	loginMiddlewareOAuth,
+	async (req: Request, res: ResponseWithRequestState) => {
+		const state = res.locals;
+		const geolocation = state.pageData.geolocation;
+		const newsletters = await getUserNewsletterSubscriptions({
+			newslettersOnPage: NewsletterMap.get(geolocation) as string[],
+			ip: req.ip,
+			sc_gu_u: req.cookies.SC_GU_U,
+			request_id: state.requestId,
+			accessToken: state.oauthState.accessToken?.toString(),
+		});
+		// eslint-disable-next-line functional/immutable-data
+		state.pageData.newsletters = newsletters;
+		const html = renderer('/welcome/newsletters', {
+			pageTitle: 'Choose Newsletters',
+			requestState: state,
+		});
```

Comment:
is there a reason `mergeRequestState` isn't being used here?

Reply:
...I forgot about it. Next question